### DISCUSSION TITLE
Abbreviations are not expanding on `<space>`  but do expand on hitting `<return>`.

### DISCUSSION THREAD

**Discussion by s-a-c** [olets/zsh-abbr#157] [Q&A]
can you please help?
I'm sure I've read in the docs about the ' ' being used to delimit and expand abbreviations -- I can't seem to find it now and I've obviously done something to mess up the default behaviour.

For clarity, and as an example:
'
"eza"="eza --classify --color=always --color-scale --group-directories-first --hyperlink --icons"
"l"="eza"
'
with the above abbreviations -- I need to hit `<return>` immediately after the abbreviation, in order to get expansion and execution.
If I try to type a traiiing `<space>`, there is no expansion preview. Moreover, expansion now fails for execution.

So: `l<return>` expands and runs `eza` successfully.

BUT `l<space><return>` produces `"zsh: command not found: l"`.

**Comment by olets** [comment]
The relevant docs are here https://zsh-abbr.olets.dev/widgets-and-key-bindings.html

//

Making sure I understand: 

1. you want space to expand and then execute
2. not ony is space not doing that, but it isn't even expanding

Is that right?

Did you have that working before? Does your setup do something custom with `abbr-expand-and-space`? In the recently-released v6, that's been replaced with `abbr-expand-and-accept`. 

//

> Moreover, expansion now fails for execution.

This bit is expected. Trailing (and leading) whitespace matters in abbreviations.

**Comment by olets** [comment]
Closing because stale, and several trees back. Revive if necessary.
